### PR TITLE
Filter transform region detail

### LINKED ISSUE #3310: Filter & Transform node-health_care_region_detail_page content ## Problem Statement
Our tome-sync data doesn't match up with our templates, and we need to follow the steps [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/2835) to ensure that the data will be compatible with our current templates.

## Goal
Content from `node-health_care_region_detail_page` entities match their corresponding entities from `pages.json`

### PR DESCRIPTION
**Pull request by bkjohnson**
### Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/3310

This filters and transforms the `node-health_care_region_detail_page` entities

It looks like a lot of changes, but 15 of the files are just `paragraph.*.json` files being added that are children or grandchildren of the entity

## Testing done

Yup

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/stages/build/process-cms-exports/schemas/raw/node-health_care_region_detail_page.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      type: 'array',
+      maxItems: 1,
+      items: { $ref: 'EntityReference' },
+    },
+    field_related_links: {
+      type: 'array',
+      maxItems: 1,
+      items: { $ref: 'EntityReference' },
+    },
+    field_table_of_contents_boolean: { $ref: 'GenericNestedBoolean' },
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/schemas/raw/node-health_care_region_detail_page.js]: Data type in the name. Cool. Any idea what it does, out of curiosity?

File: src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        },
+      },
+      required: ['entity'],
+    },
+    fieldRelatedLinks: {
+      oneOf: [
+        { $ref: 'transformed/paragraph-list_of_link_teasers' },
+        { type: 'null' },
+      ],
+    },
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js]: :100:

File: src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          properties: {
+            entityLabel: { type: 'string' },
+            fieldNicknameForThisFacility: { type: 'string' },
+            title: { type: 'string' },
+          },
+          required: ['entityLabel', 'fieldNicknameForThisFacility', 'title'],
+        },
+      },
+      required: ['entity'],
+    },
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js]: Like we discussed in Slack, I think this is find to use the full entity. It'll have "extra" properties available in the transformers, but it'll be cleaner. One fewer exception to deal with, y'know?

File: src/site/stages/build/process-cms-exports/transformers/node-health_care_region_detail_page.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  utcToEpochTime,
+  createMetaTagArray,
+} = require('./helpers');
+
+const transform = entity => ({
+  entityType: 'node',
+  entityBundle: 'health_care_region_detail_page',
+  title: getDrupalValue(entity.title),
+  changed: utcToEpochTime(getDrupalValue(entity.changed)),
+  entityPublished: getDrupalValue(entity.moderationState) === 'published',
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/transformers/node-health_care_region_detail_page.js]: I just made [this](https://github.com/department-of-veterans-affairs/vets-website/pull/11497/files?file-filters[]=.js#diff-1d5006db7b1dc3569634f6f4a03e526fR151-R153) in my last PR, for what it's worth. To limit our magic string usage.

File: src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      // TODO: Add breadcrumb here
+      properties: {
+        path: { type: 'string' },
+      },
+      required: ['path'],
+    },
+    fieldAlert: { type: ['string', 'null'] },
+    fieldContentBlock: {
+      type: 'array',
+      items: { $ref: 'Paragraph' },
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js]: Er....not `BlockContent`? :confused: 

Naming is hard...

File: src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    },
+    fieldFeaturedContent: { type: 'array' },
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js]: Of anything in particular that you could tell?

File: src/site/stages/build/process-cms-exports/transformers/node-health_care_region_detail_page.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  fieldOffice: {
+    entity: {
+      entityLabel: entity.fieldOffice[0].entity.entityLabel,
+      fieldNicknameForThisFacility:
+        entity.fieldOffice[0].entity.fieldNicknameForThisFacility,
+      title: entity.fieldOffice[0].entity.title,
+    },
+  },
+  fieldRelatedLinks:
+    entity.fieldRelatedLinks.length > 0 ? entity.fieldRelatedLinks[0] : null,
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/transformers/node-health_care_region_detail_page.js]: Alternatively:
```suggestion
    entity.fieldRelatedLinks[0] || null,
```

File: src/site/stages/build/process-cms-exports/transformers/node-health_care_region_detail_page.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  fieldIntroText: getDrupalValue(entity.fieldIntroText),
+  fieldOffice: {
+    entity: {
+      entityLabel: entity.fieldOffice[0].entity.entityLabel,
+      fieldNicknameForThisFacility:
+        entity.fieldOffice[0].entity.fieldNicknameForThisFacility,
+      title: entity.fieldOffice[0].entity.title,
+    },
+  },
+  fieldRelatedLinks:
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/transformers/node-health_care_region_detail_page.js]: Like I said above, we can probably get away with `fieldOffice: entityFieldOffice[0] || null,`.

Honestly, it isn't the end of the world either way, but I think the less clever we can be about it, the better off we'll be down the road.

File: src/site/stages/build/process-cms-exports/transformers/node-page.js [hunk 9]
```diff
@@ -24,7 +29,7 @@ function pageTransform(entity) {
 
     fieldIntroText: getDrupalValue(fieldIntroText),
     fieldDescription: getDrupalValue(fieldDescription),
-    changed: new Date(getDrupalValue(changed)).getTime() / 1000,
+    changed: utcToEpochTime(getDrupalValue(changed)),
```
Comment (cvalarida) [inline src/site/stages/build/process-cms-exports/transformers/node-page.js]: Ooh, good call. :+1:

**Review by cvalarida — COMMENTED** [review]
For what it's worth, those paragraph entities are grandchildren or lower; the scaffolder copies all direct children over.

It looks good, but I had a few questions before I give it the thumbs up.

File: src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js (hunk 6 above)
Comment (bkjohnson) [reply src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js]: :man_facepalming: I meant to fix this before I marked it ready - thank you.

File: src/site/stages/build/process-cms-exports/schemas/raw/node-health_care_region_detail_page.js (hunk 1 above)
Comment (bkjohnson) [reply src/site/stages/build/process-cms-exports/schemas/raw/node-health_care_region_detail_page.js]: It looks like it [controls whether or not some DOM nodes get rendered](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/layouts/health_care_region_detail_page.drupal.liquid#L29-L34), and I'm hoping that something else comes along later to fill in the details.

File: src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js (hunk 5 above)
Comment (bkjohnson) [reply src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js]: I don't think I understand what you mean.  In the raw schema `field_content_block` is an array of `paragraph` nodes:

image

Comment (cvalarida) [reply src/site/stages/build/process-cms-exports/schemas/transformed/node-health_care_region_detail_page.js]: Yeah, that's fine. I just think `field_content_block` is an unfortunate name for a block of `paragraph-*`s and not `block_content-*`s. Nothing for you to do.

**Review by cvalarida — APPROVED** [review]
:shipit:

**Pull request merged**